### PR TITLE
My proposition for project documentation

### PR DESCRIPTION
**Pull request by Eiji7**
My proposition for #84:
1. added `ex_doc` to project dependencies
2. added `/doc` to .gitignore file
3. added docs information with alias `README.md` file as main page

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -15,8 +15,8 @@ defmodule PhoenixEcto.Mixfile do
 
      # Docs
      name: "Phoenix/Ecto",
-     docs: [source_ref: "v#{@version}",
-            source_url: "https://github.com/phoenixframework/phoenix_ecto"]]
```
Comment (josevalim) [inline mix.exs]: Don't we need those attributes still?

Comment (Eiji7) [inline mix.exs]: @josevalim: I generated docs successfully without them. I do not used these attributes before, but these attributes describes a source of docs and we want to generate docs (not link to outside source), so we do not need external docs source. Please correct me if I'm wrong.

**Comment by josevalim** [comment]
They are important for generating links to the proper source on released versions. Please do not remove them. :)

**Comment by Eiji7** [comment]
@josevalim: ok, sorry
I restored them

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**